js/src/components/contact-information/phone-number-card/useCountdown.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback, useRef } from '@wordpress/element';
+
+/**
+ * @typedef {Object} Countdown
+ * @property {number} second The current remaining seconds of countdown.
+ * @property {number} callCount The calling count of `startCountdown` function.
+ * @property {(countdownSecond: number) => void} startCountdown Start the countdown with the given starting seconds.
+ */
+
+/**
+ * Get the corresponding countdown states and its start function by given `handle`.
+ * The states would be updated every 1000 ms interval till counting down to 0.
+ *
+ * @param {string} handle The countdown handle.
+ * @return {Countdown} The corresponding countdown states and its start function.
+ */
+export default function useCountdown( handle ) {
+	const timerRef = useRef( {} );
+	const [ second, setSecond ] = useState( 0 );
+
+	timerRef.current.usingHandle = handle;
+	timerRef.current[ handle ] = timerRef.current[ handle ] || { callCount: 0 };
```

Comment:
💅 I think using `handle` and storing the timers inside `timerRef.current[ handle ]` is making things more complicated. It would be simpler to make `useCountdown` deal with just one timer.

Reply:
I implemented it in this way at first, but as the internal state of the `<VerifyPhoneNumberContent>` component became more and more complex, I tried to move the `handle` (method) here to reduce a little complexity of `<VerifyPhoneNumberContent>`.

I thought it looks elegant when using the `useCountdown` hook with `handle`, though there's a little more complexity encapsulated inside.

Reply:
> I thought it looks elegant when using the `useCountdown` hook with `handle`

Another problem I thought about is that in case we want to use only one fixed timer, then we would need to do it like:

```
const timer = useCountdown('one')
```

Technically we could still do `useCountdown()` (with `undefined`) but then things would start to get a bit weird in the implementation. 😅 

Anyway, maybe this is something we can think about and refactor later; if it works now, then we are good. 

> though there's a little more complexity encapsulated inside.

Yeah. 😅